src/main/scala/scorex/core/NodeViewModifier.scala

```diff
@@ -30,13 +30,15 @@ trait EphemerealNodeViewModifier extends NodeViewModifier
   * have identifiers of the some length fixed with the ModifierIdSize constant
   */
 object NodeViewModifier {
-  private val DefaultIdSize = 32 // in bytes, TODO: should we use type Byte?
+  private val DefaultIdSize: Byte = 32 // in bytes
 
   val ModifierIdSize: Int = Try(ConfigFactory.load().getConfig("app").getInt("modifierIdSize")).getOrElse(DefaultIdSize)
 }
 
 
 trait PersistentNodeViewModifier extends NodeViewModifier {
+  // TODO remove. parentId is not nessesary for persistent modifiers, e.g. most of persistent modifiers in Ergo
+  // TODO (BlockTransactions, ADProofs, PoPoWProofs, ...) does not have parent id
```

Comment:
We should write a definition, what is `PersistentNodeViewModifier`. If it is just a modifier, that should be persisted in a database, it might be possible to have some, that are not ordered in a chain or a DAG and do not have parent.

Reply:
Persistent node view modifier is a part of replicated log of state transformations. The log should be deterministic and ordered to get a deterministic state after its application to the genesis state.